### PR TITLE
Test: Dump all api-resources in a separate file

### PR DESCRIPTION
**Pull request by eloycoto**
To make it easier to debug, dump all api-resources in a separate file in
the artifact

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6932)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by ianvernon** [comment]
Build failure: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/9610/

Appears to be due to infrastructural / network connectivity issues to quay.io:
```
12:21:50     k8s1-1.13: Step 2/18 : FROM quay.io/cilium/cilium-builder:2018-11-01 as builder
12:22:06     k8s1-1.13: Get https://quay.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
12:22:06     k8s1-1.13: Makefile:141: recipe for target 'docker-image-no-clean' failed
12:22:06     k8s1-1.13: make: *** [docker-image-no-clean] Error 1
12:22:06 The SSH command responded with a non-zero exit status. Vagrant
12:22:06 assumes that this means the command failed. The output for this command
12:22:06 should be in the log above. Please read the output to determine what
12:22:06 went wrong.
```

**Comment by ianvernon** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -1648,11 +1648,18 @@ func (kub *Kubectl) GatherLogs() {
...
 	kub.GeneratePodLogGatheringCommands(reportCmds)
 
+	res := kub.Exec(fmt.Sprintf(`%s api-resources | grep -v "^NAME" | awk '{print $1}'`, KubectlCmd))
+	if res.WasSuccessful() {
+		for _, line := range res.ByLines() {
+			key := fmt.Sprintf("%s get %s --all-namespaces -o wide", KubectlCmd, line)
+			reportCmds[key] = fmt.Sprintf("api-resource-%s.txt", line)
+		}
+	}
+
```
Comment (ianvernon) [inline test/helpers/kubectl.go:1682]: Log an error to console / log file to indicate what the output of `res` was if it was not successful.

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

File: test/helpers/kubectl.go (hunk 1 above)
Comment (ianvernon) [reply test/helpers/kubectl.go:1682]: Minor typo - api-resoureces. Not blocking merging though, can be fixed later.

**Pull request merged**